testing/text_index_schema_test.cc

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2025, valkey-search contributors
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD 3-Clause
+ *
+ */
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "src/indexes/text/text_index.h"
+#include "src/indexes/text.h"
+
+namespace valkey_search {
+namespace indexes {
+namespace {
+
+class TextIndexSchemaTest : public ::testing::Test {
+ protected:
+  void SetUp() override {}
+  void TearDown() override {}
+};
+
+TEST_F(TextIndexSchemaTest, BasicFieldNumbering) {
+  // Create a TextIndexSchema instance
+  auto text_index_schema = std::make_shared<text::TextIndexSchema>();
+  
+  // Initially, no text fields allocated
+  EXPECT_EQ(text_index_schema->num_text_fields_, 0);
+
+  // Manually allocate field numbers and verify sequential allocation
+  EXPECT_EQ(0, text_index_schema->AllocateTextFieldNumber());
+  EXPECT_EQ(1, text_index_schema->AllocateTextFieldNumber());
+  EXPECT_EQ(2, text_index_schema->AllocateTextFieldNumber());
+  EXPECT_EQ(3, text_index_schema->num_text_fields_);
+}
```

Comment:
Do you think we need this test? The one below already covers the basic incrementing functionality and in general I don't think unit testing a function that simply returns and increments an integer provides much value